resources/RegexGrammar.pp

```diff
@@ -57,8 +59,8 @@
 %token class:literal             \\.|.|\n
 
 // Internal options.
-// See https://www.regular-expressions.info/refmodifiers.html
-%token  internal_option          \(\?([imsxnJUX^]|xx)?-?([imsxnJUX^]|xx)\)
+// See https://www.php.net/manual/en/regexp.reference.internal-options.php
+%token  internal_option          \(\?[imsxnJUX]*-?[imsxnJUX]+\)
```

Comment:
You removed `(?^)` support, but it is supported by PHP as per https://3v4l.org/BPeFg and regular-expressions.info

Aside from that, I am not sure anymore why I only made it one modifier max and no `+`. I think there was a reason but I cannot find it anymore and it seems to be supported so I guess I am 👍🏻 for the change overall.

Note that when ^ is used, you cannot remove modes as it resets them (see how the first one fails to compile here https://3v4l.org/5Snv2) so probably this should be `\(\?(?:[imsxnJUX]*-?|\^)[imsxnJUX]+\)` to be strict. Or just `\(\?[imsxnJUX]*-?[imsxnJUX^]+\)` to keep it simpler. I don't mind too much as this is a super fringe feature, and regex compilation would fail before anyway if it's invalid so we probably don't need to worry too much about invalid regexes here.

Reply:
done - https://github.com/phpstan/phpstan-src/pull/3488#discussion_r1778185558